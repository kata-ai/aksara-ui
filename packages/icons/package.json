{
  "name": "@aksara-ui/icons",
  "version": "1.0.0-canary.12",
  "description": "React icon library for the Aksara design system.",
  "sideEffects": false,
  "main": "dist/index.js",
  "module": "dist/icons.esm.js",
  "typings": "dist/index.d.ts",
  "aksara:src": "./src/index.ts",
  "repository": "https://github.com/kata-ai/wicara.git",
  "author": "Kata.ai <dev@kata.ai>",
  "license": "Apache-2.0",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsdx build --transpileOnly --tsconfig tsconfig.build.json",
    "clean": "rimraf dist",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
    "prepublish": "yarn build",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose",
    "validate": "yarn lint && yarn type-check"
  },
  "dependencies": {
    "clsx": "^1.1.1"
  },
  "peerDependencies": {
    "react": "^16.10.0 || ^17.0.0",
    "react-dom": "^16.10.0 || ^17.0.0"
  },
  "devDependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  }
}
