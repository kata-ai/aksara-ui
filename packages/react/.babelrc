{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": ["last 2 chrome versions", "last 2 firefox versions"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "babel-plugin-styled-components"
  ],
  "env": {
    "test": {
      "plugins": ["@babel/plugin-transform-modules-commonjs"]
    }
  }
}
