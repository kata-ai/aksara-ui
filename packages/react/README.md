# @aksara-ui/react

> React implementation of Aksara Design System, from Kata.ai.

This package is part of the Aksara monorepo. Please read the README in [the Aksara repository](https://github.com/kata-ai/aksara-ui) for detailed documentation about this package.
