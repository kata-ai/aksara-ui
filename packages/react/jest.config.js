module.exports = {
  collectCoverageFrom: [
    'src/**/*.{ts,tsx}',
    '!src/**/*.stories.{ts,tsx}',
    '!src/**/*.story.{ts,tsx}',
    '!src/**/index.{ts,tsx}',
    '!src/stories/**/*.{ts,tsx}',
    '!src/utils/storybook/**/*.{ts,tsx}',
    '!src/index.{ts,tsx}',
  ],
  coverageReporters: ['json', 'lcov'],
  moduleDirectories: ['node_modules', '<rootDir>/src'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json'],
  modulePathIgnorePatterns: ['lib', 'lib-es6'],
  setupFilesAfterEnv: ['<rootDir>/test/setupTests.ts'],
  testPathIgnorePatterns: ['<rootDir>/dist/'],
  transform: {
    '^.+\\.(js|jsx|ts|tsx|mjs)$': 'babel-jest',
    '^.+\\.css$': '<rootDir>/config/jest/cssTransform.js',
    '^(?!.*\\.(js|jsx|ts|tsx|mjs|css|json)$)': '<rootDir>/config/jest/fileTransform.js',
  },
};
