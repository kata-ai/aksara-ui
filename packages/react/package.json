{
  "name": "@aksara-ui/react",
  "version": "1.0.1-canary.4",
  "description": "React implementation of Aksara Design System, from Kata.ai.",
  "main": "dist/index.js",
  "module": "dist/core.esm.js",
  "typings": "dist/index.d.ts",
  "aksara:src": "./src/index.ts",
  "repository": "https://github.com/kata-ai/wicara.git",
  "author": "Kata.ai <dev@kata.ai>",
  "license": "Apache-2.0",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsdx build --transpileOnly --tsconfig tsconfig.build.json",
    "build-storybook": "build-storybook --quiet",
    "clean": "rimraf dist",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
    "prepublish": "yarn build",
    "report-coverage": "codecov -f ./coverage/*.json",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose",
    "storybook": "start-storybook -p 9001",
    "storybook:build": "build-storybook --quiet",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "type-check": "tsc --emitDeclarationOnly"
  },
  "dependencies": {
    "@aksara-ui/core": "^1.0.1-canary.1",
    "@popperjs/core": "^2.10.2",
    "@reach/accordion": "^0.16.1",
    "@reach/visually-hidden": "^0.16.0",
    "@styled-system/css": "^5.1.5",
    "@styled-system/should-forward-prop": "^5.1.5",
    "@styled-system/theme-get": "^5.1.2",
    "@types/styled-system": "^5.1.13",
    "@types/styled-system__css": "^5.0.16",
    "@types/styled-system__theme-get": "^5.0.1",
    "clsx": "^1.1.1",
    "csstype": "^3.0.9",
    "deepmerge": "^4.2.2",
    "focus-trap": "^6.6.0",
    "polished": "^4.1.3",
    "react-fast-compare": "^3.2.0",
    "react-popper": "^2.2.5",
    "react-popper-tooltip": "^4.3.0",
    "react-transition-group": "^4.4.2",
    "styled-system": "^5.1.4",
    "tslib": "^2.3.1"
  },
  "peerDependencies": {
    "@aksara-ui/icons": "^1.0.0-canary.0",
    "react": "^16.10.0 || ^17.0.0",
    "react-dom": "^16.10.0 || ^17.0.0",
    "styled-components": "^5.1.0"
  },
  "devDependencies": {
    "@aksara-ui/icons": "^1.0.0",
    "@babel/core": "^7.15.0",
    "@storybook/addon-actions": "^6.3.7",
    "@storybook/addon-essentials": "^6.3.7",
    "@storybook/addon-knobs": "^6.3.0",
    "@storybook/addon-links": "^6.3.7",
    "@storybook/react": "^6.3.7",
    "@storybook/theming": "^6.3.7",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@types/jest": "^26.0.24",
    "@types/react-transition-group": "^4.4.2",
    "@types/styled-components": "^5.1.12",
    "babel-loader": "^8.2.2",
    "babel-plugin-styled-components": "^1.13.2",
    "babel-preset-react-app": "^10.0.0",
    "codecov": "^3.8.3",
    "copee": "^1.0.6",
    "jest": "^26.6.3",
    "jest-styled-components": "^7.0.5",
    "react": "^17.0.2",
    "react-docgen-typescript-loader": "^3.6.0",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.0"
  }
}
