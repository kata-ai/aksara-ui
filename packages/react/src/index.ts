export * from './aksara-provider';
export * from './components';
export * from './layout';
export * from './theme';
export * from './typography';
export * from '@aksara-ui/helpers';
