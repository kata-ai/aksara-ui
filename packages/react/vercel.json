{
  "version": 2,
  "headers": [
    {
      "source": "/static/(.*)",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "public, max-age=0, must-revalidate"
        }
      ]
    },
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "Cache-Control",
          "value": "s-maxage=0"
        }
      ]
    }
  ]
}
