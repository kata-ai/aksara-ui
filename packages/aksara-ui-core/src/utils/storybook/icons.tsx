/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { toClipboard } from 'copee';

import { Box, Card, Tooltip, UnstyledButton, notification } from '../..';

export const IconGrid: React.FC = ({ children }) => {
  return (
    <Box
      mt="lg"
      mb="xl"
      display="grid"
      gridTemplateColumns="repeat(auto-fill, minmax(calc(1116px / 12 - 24px), 1fr))"
      gridGap="24px"
    >
      {children}
    </Box>
  );
};

export const copyText = (text: string) => {
  const success = toClipboard(text);

  if (success) {
    notification.toaster({
      status: 'success',
      message: `Successfully copied \`${text}\` to clipboard.`,
      dismissible: true,
      dismissAfter: 5000,
    });
  }
};

// TODO: Maybe use Suspense + lazy load these icons by path?
export const IconDetail: React.FC<{ name: string }> = ({ name, children }) => {
  return (
    <Tooltip style={{ display: 'flex', flexDirection: 'column' }} content={name}>
      <UnstyledButton onClick={() => copyText(name)}>
        <Card elevation={1} width="100%">
          <Box
            display="flex"
            flexDirection="column"
            alignItems="center"
            justifyContent="center"
            textAlign="center"
            height={71}
            p="sm"
          >
            <Box width={24} height={24}>
              {children}
            </Box>
          </Box>
        </Card>
      </UnstyledButton>
    </Tooltip>
  );
};
