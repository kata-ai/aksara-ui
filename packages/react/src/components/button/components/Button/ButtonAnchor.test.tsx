import * as React from 'react';
import { render } from '@testing-library/react';
import { IconDocAdd } from '@aksara-ui/icons';

import { ButtonAnchor } from './index';

describe('components/Button', () => {
  describe('<ButtonAnchor />', () => {
    test('renders correctly', () => {
      const { container } = render(<ButtonAnchor>example</ButtonAnchor>);

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders colors correctly by variant', () => {
      const { container } = render(<ButtonAnchor variant="primary">test button</ButtonAnchor>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders block buttons correctly', () => {
      const { container } = render(<ButtonAnchor block>test button</ButtonAnchor>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders icons correctly', () => {
      const { container } = render(
        <ButtonAnchor size="lg" icon={IconDocAdd} iconPosition="right">
          test button
        </ButtonAnchor>
      );
      const icon = container.querySelector('svg');

      expect(icon).toBeInTheDocument();
    });

    test('renders selected styles correctly', () => {
      const { container } = render(
        <ButtonAnchor variant="outline" selected>
          test button
        </ButtonAnchor>
      );

      expect(container.firstChild).toHaveClass('selected');
    });
  });
});
