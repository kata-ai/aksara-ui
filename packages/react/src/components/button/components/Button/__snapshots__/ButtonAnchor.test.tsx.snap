// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button <ButtonAnchor /> renders block buttons correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<a
  class="c0"
>
  <span
    class="Text-sc-18vu504-0"
  >
    test button
  </span>
</a>
`;

exports[`components/Button <ButtonAnchor /> renders colors correctly by variant 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<a
  class="c0"
>
  <span
    class="Text-sc-18vu504-0"
  >
    test button
  </span>
</a>
`;
